userguide/aws/how-tos/add-a-certificate.adoc

```diff
@@ -0,0 +1,161 @@
+
+The remainder of this document has three parts:
+
+. <<step0, Preliminary Concepts>>
+. <<step1, Obtaining a Certificate>>
+. <<step2, Testing and Verification>>
+
+[[step0]]
+== Preliminary concepts
+
+=== CNAMEs
+
+A CNAME, or *Canonical Name*, is an identifier that links your CA to the resources that you are protecting with your certificate.CNAMEs are discussed in https://docs.killbill.io/latest/aws/explanations/https.html[HTTPS and Certificates]. Further instructions for setting up your required CNAMEs are given here.
+
+=== Protocol and Port Setup
+
+Your AWS security group by default provides rules giving access to Kaui and Kill Bill using the HTTP protocol and a specified port number. For HTTPS you will need additional rules with the new protocol and a new port number.
+
+=== Validation
+
+When you create a new certificate you will need to validate your certificate. This identifies the trust hierarchy for the certificate. You may need to setup a temporary CNAME for this purpose, and you may need to provide temporary access via an extra port in your security group.
+
+[[step1]]
+== Step 1: Obtaining a Certificate
+
+There are many ways to obtain a certificate that you can use to protect a particular resource. Here we will describe two that may be of use in particular Kill Bill scenarios running on AWS. These are:
+
+1. Using the AWS Certificate Manager
+2. Using the free CA *Let's Encrypt*
+
+=== Using the AWS Certificate Manager (ACM)
+
+The AWS Certificate Manager (ACM) offers a straightforward way to provide SSL/TLS protection for AWS resources such as load balancers. This is recommended when using the multi-tier or cloud formation installation options. Each of these options makes use of the AWS Elastic Load Balancers (ELBs). For these cases there is no extra cost for using ACM certificates. 
+
+==== 1. Request a Certificate
+
+If you are setting up a new multi-tier or CloudFormation implementation, choose the HTTPS option for your load balancer. You will come to a page that asks for your certificate. click on *Request a new Certificate from ACM*. 
+
+Otherwise, to add HTTPS security to an existing implementation, select *Security, Identity, and Compliance*, then *Certificate Manager* from the Services menu. In this case you will need to edit your load balancer(s) later on.
+
+In either case you will be taken to the main page of the ACM. Initially this will probably show that you have no certificates. On the left menu, click *Request a Certificate*.
+
+The next page will give you the option to request a public or private certificate. The private option may be grayed out. Click *Request a Public Certificate*. The page that appears will ask you to specify several parameters, including one or more domain names and a validation method.
+
+image::https://github.com/killbill/killbill-docs/raw/v3/userguide/assets/aws/ELB-request-certificate.png[align=center]
+
+==== 2. Setup a CNAME
```

Comment:
Should this be changed to "Set up a validation CNAME"?

Reply:
Perhaps it should be "Set up CNAMEs". ACM requires both a validation CNAME and a regular CNAME. It seems best to create them both at the same time.